python/services/monitorservice/dmod/monitorservice/service.py

```diff
@@ -1,14 +1,584 @@
+
+        for i in jobs_of_interest_subset:
+            if not isinstance(i, str):
+                raise RuntimeError('Invalid metadata with non-string id value in \'jobs-of-interest\' list')
+
+        return list(jobs_of_interest_subset)
+
+    @abstractmethod
+    async def communicate_change(self, change: MonitoredChange):
+        """
+        Inform the associated client of this monitored change.
+
+        Inform some client, associated by the connection id contained in the change object, of the encapsulated change.
+
+        Parameters
+        ----------
+        change: MonitoredChange
+            Object containing a change found in monitoring and the id of a client needing to know about the change.
+        """
         pass
 
-    def run():
+    async def exec_monitoring(self):
+        """
+        Async task performing repeating, regular monitoring tasks within service, and queuing monitored changes that
+        are of interest to parties with current connections to the service.
+
+        See Also
+        ----------
+        ::method:`run_monitor_check`
+        """
         while True:
-            print("Not Implemented")
-            break
+            self.run_monitor_check()
+            await sleep(60)
+
+    @abstractmethod
+    def get_connection_object(self, connection_id: str):
+        """
+        Get the object encapsulating some identified connection.
+
+        Parameters
+        ----------
+        connection_id: str
+            The identifier for the connection.
+
+        Returns
+        -------
+        The desired connection object.
+        """
+        pass
+
+    def handle_connection_begin(self, message: str) -> Tuple[str, Optional[MetadataMessage], MetadataResponse]:
+        """
+        Helper method for tasks to do at the beginning of a connection.
+
+        Parameters
+        ----------
+        message: str
+            The raw JSON string included in the first message, expected to be a serialized ::class:`MetadataMessage`.
+
+        Returns
+        -------
+        Tuple[str, Optional[MetadataMessage], MetadataResponse]
+            The generated id for this connection, a metadata message object deserialized from the message (if possible),
+            and a metadata response object with deals on whether connect is successful.
+        """
+        connection_id = str(uuid.uuid4())
+        # Process the incoming connection message
+        metadata_obj, connect_success, response_txt = self._proc_connect_json(message, connection_id)
+        # TODO: any other steps to initialize connection (e.g., need auth or session key)?
+        # If things are valid, extract the (potentially implied) jobs of interest
+        if connect_success:
+            try:
+                self.jobs_of_interest_by_connection[connection_id] = self._proc_metadata_jobs_of_interest(metadata_obj)
+            # Consider an exception during this to override connection success
+            except RuntimeError as re:
```

Comment:
The only error I was expecting could be raised in this block is a _RuntimeError_ from _MonitorService_'s `_proc_metadata_jobs_of_interest()` function (if there is something I am missing, let me know).   And _RuntimeError_ there seemed appropriate, given the situations in the function that would raise it.  I could create a custom error type and use it instead; it just seemed overly specialized.  If you think we need it though, we can add it.

Reply:
`RuntimeError` is probably sufficient, but if we wanted "graceful" protocol shutdown, then I think any exception would be appropriate. 

Since we aren't trying to recover from a specific error, a custom error type isn't going to be too useful.